pkg/token/execCredentialWriterAPIVersion_test.go

```diff
@@ -0,0 +1,69 @@
+package token
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"k8s.io/client-go/pkg/apis/clientauthentication"
+)
+
+
+func TestExecCredentialWriterAPIVersion() (t *testing.T) {
+	testData1 := []struct {
+		name               string
+		execInfoEnvTest    string
+		stringBufferTest   bytes.Buffer
+		expectedAPIVersion string
+	}{
+		{
+			name: "KUBERNETES_EXEC_INFO is empty",
```

Comment:
typical test case should describe the scenario and the expected outcome

Reply:
As the test logic for this is relatively easy so I put these in t.Run() func.